components/eventing-controller/controllers/subscription/nats/integration_test.go

```diff
@@ -0,0 +1,837 @@
+				},
+			},
+			shouldTestDeletion: true,
+		},
+
+		{
+			name: "valid sink; with port",
+			subscriptionOpts: []reconcilertesting.SubscriptionOpt{
+				reconcilertesting.WithFilter("", reconcilertesting.OrderCreatedEventTypeNotClean),
+				reconcilertesting.WithSinkURL(validSinkURL(ens, ":8080")),
+			},
+			expect: expect{
+				k8sSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.HaveCondition(reconcilertesting.DefaultCondition("")),
+					reconcilertesting.HaveSubscriptionReady(),
+				},
+			},
+			shouldTestDeletion: true,
+		},
+
+		{
+			name: "valid sink; with port and endpoint",
+			subscriptionOpts: []reconcilertesting.SubscriptionOpt{
+				reconcilertesting.WithFilter("", reconcilertesting.OrderCreatedEventTypeNotClean),
+				reconcilertesting.WithSinkURL(validSinkURL(ens, ":8080", "/myEndpoint")),
+			},
+			expect: expect{
+				k8sSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.HaveCondition(reconcilertesting.DefaultCondition("")),
+					reconcilertesting.HaveSubscriptionReady(),
+				},
+			},
+			shouldTestDeletion: true,
+		},
+
+		{
+			name: "empty protocol, protocol setting and dialect",
+			subscriptionOpts: []reconcilertesting.SubscriptionOpt{
+				reconcilertesting.WithFilter("", reconcilertesting.OrderCreatedEventTypeNotClean),
+				reconcilertesting.WithSinkURLFromSvc(ens.subscriberSvc),
+			},
+			expect: expect{
+				k8sSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.HaveCondition(reconcilertesting.DefaultCondition("")),
+				},
+				natsSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.BeNotNil(),
+					reconcilertesting.BeValid(),
+					reconcilertesting.HaveSubject(reconcilertesting.OrderCreatedEventType),
+				},
+			},
+			shouldTestDeletion: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			subscription, subscriptionName := createSubscription(ctx, g, ens, tc.subscriptionOpts...)
+			testSubscriptionOnK8s(ctx, g, ens, subscription, subscriptionName, tc.expect.k8sSubscription...)
+			testEventsOnK8s(ctx, g, ens, tc.expect.k8sEvents...)
+			testSubscriptionOnNATS(g, ens, subscriptionName, tc.expect.natsSubscription...)
+			testDeletionOnK8s(ctx, g, ens, subscription, tc.shouldTestDeletion)
+		})
+	}
+}
+
+//TestChangeSubscription tests if existing subscriptions are reconciled properly after getting changed.
+func TestChangeSubscription(t *testing.T) {
+	ctx := context.Background()
+	g := gomega.NewGomegaWithT(t)
+	ens := setupTestEnsemble(ctx, reconcilertesting.EventTypePrefix, g, 4224, 7074)
+	defer ens.cancel()
+
+	var testCases = []struct {
+		name               string
+		subscriptionOpts   []reconcilertesting.SubscriptionOpt
+		expectBefore       expect
+		changeSubscription func(subscription *eventingv1alpha1.Subscription)
+		expectAfter        expect
+		shouldTestDeletion bool
+	}{
+		{
+			name: "clean event types; add filters to subscription without filters",
```

Comment:
I am trying to find a better name for the test. What is does it adding unclean event types and expecting that they are cleaned later in the status. The current name does not reflect this 🤔 However it is hard to find a good name 😃

Some ideas:

```suggestion
			name: "adding unclean event types eventually leads to cleaned event types",
```

```suggestion
			name: "eventually, adding unclean event types to the spec leads to cleaned event types in the status",
```

```suggestion
			name: "unclean event types in the spec are eventually clean inside the status",
```

Reply:
I just renamed them to `cleanEventTypes` since this is what they try to test. the fact that you didn't get that is proof, that my naming was terrible :D